docs/guides/migration_guide_vpcgw_v2.md

```diff
@@ -0,0 +1,167 @@
+---
+page_title: "Moving a Public Gateway from Legacy mode to IPAM mode, for v2 compatibility"
+---
+
+# Moving a Public Gateway from Legacy mode to IPAM mode
+
+This guide explains how to move a Public Gateway from [Legacy mode](https://www.scaleway.com/en/docs/public-gateways/concepts/#ipam) to IPAM mode. Only gateways in IPAM mode will be compatible with the new v2 of the Public Gateways API. v1 of the API is deprecated, and will be removed later this year.
+In the legacy setup, DHCP and DHCP reservations are managed with dedicated resources and referenced in the gateway network.
+In IPAM mode, these functionalities are managed by Scaleway IPAM.
+In 2023, DHCP functionality was moved from Public Gateways to Private Networks, DHCP resources are now no longer needed.
+
+Note:
+Trigger the move from Legacy mode to IPAM mode by setting the `move_to_ipam` flag on your Public Gateway resource.
+You can do this via the Terraform configuration or by using the Scaleway CLI/Console.
```

Comment:
Maybe put the command here with a mention of the minimal CLI version that got this feature